### PR TITLE
Add call to container/{id}/wait

### PR DESCRIPTION
**Pull request by ry99**
Adds a `container.waitOn()` function to wait for a container to complete. I did not add an integration test because the integration tests do not seem to work on my dev machine (https://github.com/amihaiemil/docker-java-api/issues/303).

### CODE REVIEW

**Comment by amihaiemil** [comment]
@ry99 Thank you for your PR. Could you at least add a simple unit tests? See how other unit tests are implemented, it should be very similar :)

File: src/main/java/com/amihaiemil/docker/Container.java [hunk 1]
```diff
@@ -151,5 +151,18 @@ void remove(final boolean volumes, final boolean force, final boolean link)
...
+
+
+    /**
+     * Waits on this container.
+     * @param state The state to wait for. One of "not-running"
+     * (the default if null), "next-exit", or "removed"
+     * @see <a href="https://docs.docker.com/engine/api/v1.35/#operation/ContainerWait">
+     * Wait Container</a>.
+     * @throws IOException If something goes wrong.
+     *  the expected one (200).
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/Container.java:163]: @ry99 I think this javadoc is broken :D

File: src/main/java/com/amihaiemil/docker/RtContainer.java [hunk 2]
```diff
@@ -230,4 +231,29 @@ public void unpause() throws IOException {
...
+    public int waitOn(final String state) throws IOException {
+        UncheckedUriBuilder urib = new UncheckedUriBuilder(
+            this.baseUri.toString().concat("/wait")
+        );
+        if(null == state || state.isEmpty()){
+            urib.addParameter("condition", state);
+        }
+
+        final HttpPost waiter = new HttpPost(urib.build());
+        waiter.setHeader(new BasicHeader("Content-Type", "application/json"));
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/RtContainer.java:245]: @ry99 I don't think you need to add this header here. I don't remember adding it in other parts of the library.

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) not enough funds available in the project, can't set budget of job [#329](https://github.com/amihaiemil/docker-java-api/issues/329), see [§21](http://www.zerocracy.com/policy.html#21); @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) will get no money on completion; in order to fix that, add funds to the project and assign the job again

File: src/main/java/com/amihaiemil/docker/RtContainer.java (hunk 2 above)
Comment (ry99) [reply src/main/java/com/amihaiemil/docker/RtContainer.java]: removed

File: src/main/java/com/amihaiemil/docker/Container.java (hunk 1 above)
Comment (ry99) [reply src/main/java/com/amihaiemil/docker/Container.java:163]: Javadoc passes the checkstyle rules now. sorry

**Comment by amihaiemil** [comment]
@rultor try to merge

**Comment by amihaiemil** [comment]
@ry99 thanks a lot for the PR!

**Comment by amihaiemil** [comment]
@rultor release it, tag is `0.0.12`

**Comment by 0crat** [comment]
Job [#329](https://github.com/amihaiemil/docker-java-api/issues/329) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't set title

**Comment by 0crat** [comment]
Job was finished in 10 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**